notebook/notebookapp.py

```diff
@@ -802,6 +808,14 @@ def _token_default(self):
         """
     )
 
+    min_open_files_limit = Integer(4096, config=True,
```

Comment:
Feel free to ignore this suggestion, but I think the traitlet name be clearer if it were `soft_open_files_limit` or `lower_open_files_limit`. The reason I suggest this is that this is a soft bound on the maximum number of files that can be created, rather than a minimum number of files that must be opened.

Reply:
To be fair, you could call it simply `open_files_limit` as well, since in practice we wouldn't be allowing people to increase the hard limit imposed by the operating system.

Reply:
I called it `min_open_files_limit` because I only set it if the current open files limit is lower than this number (i.e., I will never decrease an open files limit to this number, only raise lower open file limits to at least this.)

https://github.com/jupyter/notebook/pull/4893/files#diff-bbd307804bb6b8836da9f06b8d8a06e9R1405

That said, I also think the name is confusing, but descriptive, and would love an easier to understand name.